crates/noirc_evaluator/src/ssa/integer.rs

```diff
@@ -570,10 +570,26 @@ pub fn get_max_value(ins: &Instruction, lhs_max: BigUint, rhs_max: BigUint) -> B
         Operation::Gt => BigUint::one(),
         Operation::Lte => BigUint::one(),
         Operation::Gte => BigUint::one(),
-        Operation::And => ins.res_type.max_size(),
+        Operation::And => {
+            BigUint::from(2_u32).pow(u64::min(lhs_max.bits(), rhs_max.bits()) as u32)
+                - BigUint::one()
+        }
+        Operation::Xor | Operation::Or => {
+            BigUint::from(2_u32).pow(u64::min(lhs_max.bits(), rhs_max.bits()) as u32)
```

Comment:
Looks like the maxes for And, Xor, and Or are the same, lets group them together

Reply:
Oops, no they should not be the same...I fixed it.